README.md

```diff
@@ -41,96 +72,186 @@ pod "KlaviyoSwift"
+* Call [`registerForRemoteNotifications()`](https://developer.apple.com/documentation/uikit/uiapplication/1623078-registerforremotenotifications)
+to request a push token from APNs. This is typically done in the [`application:didFinishLaunchingWithOptions:`](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622921-application) method of your app delegate.
+* Implement the delegate method [`application:didRegisterForRemoteNotificationsWithDeviceToken`](https://developer.apple.com/documentation/appkit/nsapplicationdelegate/1428766-application)
+in your application delegate to receive the push token from APNs and register it with Klaviyo.
 
+Below is the code to do both of the above steps:
 ```swift
-KlaviyoSDK().set(email: "junior@blob.com")
-```
+import KlaviyoSwift
 
-## Push Notifications
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    KlaviyoSDK().initialize(with: "YOUR_KLAVIYO_PUBLIC_API_KEY")
 
-Implementing push notifications requires a few additional snippets of code to enable.:
+    UIApplication.shared.registerForRemoteNotifications()
 
-1. Registering users for push notifications.
-2. Sending resulting push tokens to Klaviyo.
-3. Handling when users attempt to open your push notifications.
+    return true
+}
+
+func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+    KlaviyoSDK().set(pushToken: deviceToken)
+}
+```
+#### Push Tokens And Multiple Profiles
+
+Klaviyo SDK will disassociate the device push token from the current profile whenever it is reset by calling
+`set(profile:)` or `resetProfile`. You should call `set(pushToken:)` again after resetting the currently tracked profile
+to explicitly associate the device token to the new profile.
 
-### Sending push notifications
+### Request Push Notification Permission
 
-1. Add the following code to your application wherever you would like to prompt users to register for push notifications. This is often included within `application:didFinishLaunchingWithOptions:`, but it can be placed elsewhere as well. When this code is called, ensure that the Klaviyo SDK is configured and that `set(email:)` is called. This enables Klaviyo to match app tokens with profiles in Klaviyo customers.
+Now that we have the push token, in order to send push notifications to your users, you must request permission to send push notifications.
+Add the following code to your application wherever you would like to prompt users to register for push notifications.
+This is often included within [`application:didFinishLaunchingWithOptions:`](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622921-application)
```

Comment:
Is this really a best practice...I think it's fine for now though.

Reply:
you mean don't suggest they request push permission during `didFinishLaunchingWithOptions`?

Reply:
Yah I mean it would essentially prompt when the app starts...I wonder if you could get better opt-in rates with a different strategy.

Reply:
Thanks. I updated this section to include apple docs and made it clear that app start up isn't the best location for requesting push permission.